Assets/Scripts/LevelGenerator/DungeonGenerator.cs

```diff
@@ -0,0 +1,275 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+using Random = System.Random;
```

Comment:
might be worth looking into system random vs unityengine random for this
https://answers.unity.com/questions/1151234/what-is-the-difference-between-systemrandom-and-un.html

Reply:
Already have. System random is the appropriate one to use here, since all we need from it are some random integers. Unity's random has a lot more functionality but it comes at the cost of performance. Also, like that link says using unity's random here would create a dependence between each random.Next() call due to the shared buffer.

TLDR: unity's random is slower and not truly random, but easier to use in some cases